vendor/github.com/jacobsa/gcloud/gcs/conn.go

```diff
@@ -101,10 +101,38 @@ type ConnConfig struct {
 	// responses. If nil, no logging is performed.
 	GCSDebugLogger  *log.Logger
 	HTTPDebugLogger *log.Logger
+
+	// Config parameters for the custom http client of Go Storage Client.
+	MaxConnsPerHost     int
+	MaxIdleConnsPerHost int
+	ForceAttemptHTTP2   bool
+	EnableHTTP1         bool
+	DisableKeepAlives   bool
+}
+
+// Go Client Config structure.
+type GoClientConfig struct {
+	MaxConnsPerHost     int
+	MaxIdleConnsPerHost int
+	ForceAttemptHTTP2   bool
+	EnableHTTP1         bool
+	DisableKeepAlives   bool
 }
 
+var tokenSrc oauth2.TokenSource    // Token Source for auth of custom http client of Go Storage Client.
+var goClientConfig *GoClientConfig // Contains all the config info for the Go Client.
```

Comment:
any specific reason you made these as global constants instead making it part of conn struct?